### ISSUE TITLE
Collection Hooks stops working 

### ISSUE THREAD

**Issue by namirsab** [matb33/meteor-collection-hooks#252]
Hi everyone
We have a weird issue that makes collection hooks not to run (the callbacks are not even run)
for some weird reason some times.
We have a cluster of 10 instances for our application, and some of them get "corrupted" and don't trigger the hooks any more.

Is there anything that could cause this? It's really weird, because it's not data related as in the other instances is working.

I'm trying to read the code to find a condition that would prevent the hooks to run, but I didn't find anything yet. 
Can maybe someone point me or guide me?

**Comment by evolross** [comment]
Weird.  Never seen that one.

I did have a weird bug where Collection updates inside of callbacks of Collection updates using `direct` would also fire as `direct` even when not using that directive, but there's a work-around.

This doesn't sound like that issue if the entire container gets affected. I'm assuming the actual Collection transaction still fires - e.g. the `update`, `insert`, etc. still works? Just without the hook firing?

**Comment by namirsab** [comment]
@evolross exactly, the transaction happens, but the hook is just not triggered.
